tests/tck/features/optimizer/EmbedEdgeAllPredIntoTraverseRule.feature

```diff
@@ -0,0 +1,284 @@
+      MATCH (friend)-[served:serve]->(friendTeam:team)
+      WHERE served.start_year > 2010 AND all(i in e1 where i.likeness > 1)
+      WITH DISTINCT friend, friendTeam
+      OPTIONAL MATCH (friend)<-[e2:like*2..4]-(friend2:player)<-[:like]-(friendTeam)
+      WITH friendTeam, count(friend2) AS numFriends, e2
+      WHERE e2 IS NULL OR all(i in e2 where i IS NULL)
+      RETURN
+        friendTeam.team.name AS teamName,
+        numFriends,
+        [i in e2 | i.likeness] AS likeness
+        ORDER BY teamName DESC
+        LIMIT 8
+      """
+    Then the result should be, in any order, with relax comparison:
+      | teamName        | numFriends | likeness |
+      | "Warriors"      | 0          | NULL     |
+      | "Trail Blazers" | 0          | NULL     |
+      | "Spurs"         | 0          | NULL     |
+      | "Rockets"       | 0          | NULL     |
+      | "Raptors"       | 0          | NULL     |
+      | "Pistons"       | 0          | NULL     |
+      | "Lakers"        | 0          | NULL     |
+      | "Kings"         | 0          | NULL     |
+    And the execution plan should be:
+      | id | name           | dependencies | profiling data | operator info                                                                |
+      | 28 | TopN           | 24           |                |                                                                              |
+      | 24 | Project        | 30           |                |                                                                              |
+      | 30 | Aggregate      | 29           |                |                                                                              |
+      | 29 | Filter         | 44           |                | {"condition": "($e2 IS NULL OR all(__VAR_2 IN $e2 WHERE $__VAR_2 IS NULL))"} |
+      | 44 | HashLeftJoin   | 15,43        |                |                                                                              |
+      | 15 | Dedup          | 14           |                |                                                                              |
+      | 14 | Project        | 35           |                |                                                                              |
+      | 35 | HashInnerJoin  | 53,50        |                |                                                                              |
+      | 53 | Filter         | 52           |                | {"condition": "(id($friend)!=\"Tony Parker\")"}                              |
+      | 52 | AppendVertices | 59           |                |                                                                              |
+      | 59 | Filter         | 60           |                | {"condition": "(id($person)==\"Tony Parker\")"}                              |
+      | 60 | Traverse       | 2            |                | {"filter": "((like.likeness>0) AND (like.likeness>1))"}                      |
```

Comment:
we could merge the condition into `like.likeness>1` later.

Reply:
Yep. Expression normalization is not optimized yet.